src/workerd/api/http.c++

```diff
@@ -417,6 +418,143 @@ bool Headers::inspectImmutable() {
   return guard != Guard::NONE;
 }
 
+// -----------------------------------------------------------------------------
+// serialization of headers
+//
+// http-over-capnp.capnp has a nice list of common header names, taken from the HTTP/2 standard.
+// We'll use it as an optimization.
+//
+// Note that using numeric IDs for headers implies we lose the original capitalization. However,
+// the JS Headers API doesn't actually give the application any way to observe the capitalization
+// of header names -- it only becomes relevant when serializing over HTTP/1.1. And at that point,
+// we are actually free to change the capitalization anyway, and we commonly do (KJ itself will
+// normalize capitalization of all registered headers, and http-over-capnp also loses
+// capitalization). So, it's certainly not worth it to try to keep the original capitalization
+// across serialization.
+
+// If any more headers are added to the CommonHeaderName enum later, we should be careful about
+// introducing them into serialization. We need to roll out a change that recognizes the new IDs
+// before rolling out a change that sends them. MAX_COMMON_HEADER_ID is the max value we're willing
+// to send.
+static constexpr uint MAX_COMMON_HEADER_ID =
+    static_cast<uint>(capnp::CommonHeaderName::WWW_AUTHENTICATE);
+
+static kj::Array<kj::StringPtr> makeCommonHeaderList() {
+  auto enums = capnp::Schema::from<capnp::CommonHeaderName>().getEnumerants();
+  auto builder = kj::heapArrayBuilder<kj::StringPtr>(enums.size());
+  bool first = true;
+  for (auto e: enums) {
+    if (first) {
+      // Value zero is invalid, skip it.
+      static_assert(static_cast<uint>(capnp::CommonHeaderName::INVALID) == 0);
```

Comment:
Minor nit: Having this `static_assert` inside the for loop when it doesn't seem to be testing `e` is... a bit odd.

Reply:
This `static_assert` is showing that the first enum value (zero) has the name INVALID, just to prove that the comment above is correct. I like to write `static_assert`s as close as possible to the code that is relying on whatever is asserted.

Reply:
Yep, it just looks a bit out of place since it's not testing `e` directly here. Minor issue but I can see someone coming into the code later and being a bit confused.